### PR TITLE
feat: update portfolio metadata

### LINKED ISSUE #28: API endpoint to update portfolio metadata
## Features

PUT `/portfolios/{pid}`

### Core

Request should look like:
```js
{
    "name": "new name"  // "name" or "description" or both
}
```

Above request updates portfolio `name`.

Implement Update for:
- portfolio name
- portfolio description

### PR DESCRIPTION
**Pull request by jeromeandrewong**
Closes #28 

## Features
- error handling for `pid` not found
- error handling for if both `portfolioName` and `description` not in request body
- logging

## Route
PUT: `/portfolio/<pid>`

## Demo
CleanShot 2023-11-05 at 00 25 28

### CODE REVIEW

File: src/main/java/com/backend/controller/PortfolioController.java [hunk 1]
```diff
@@ -103,6 +107,37 @@ public GetPortfolioByIdResponse getPortfolio(@PathVariable long pid) {
 		return response;
 	}
 
+	@PutMapping(path = "/portfolio/{pid}")
+	public UpdatePortfolioMetadata updatePortfolio(@PathVariable long pid, @RequestBody Map<String, String> body) {
```
Comment (DavidLHW) [inline src/main/java/com/backend/controller/PortfolioController.java:111]: should probably replace `Map<String, String>` with `UpdatePortfolioMetadataRequest`

under `./src/main/java/com/backend/request`:

```java
package com.backend.request;

import lombok.Data;
import lombok.EqualsAndHashCode;

@Data
@EqualsAndHashCode(callSuper = false)
class UpdatePortfolioMetadataRequest extends BaseRequest {
    String portfolioName;
    String description
}
```

File: src/main/java/com/backend/response/UpdatePortfolioMetadata.java [hunk 2]
```diff
@@ -0,0 +1,9 @@
+package com.backend.response;
+
+import lombok.Data;
+
+@Data
+public class UpdatePortfolioMetadata {
```
Comment (DavidLHW) [inline src/main/java/com/backend/response/UpdatePortfolioMetadata.java:6]: should follow same naming convention: `UpdatePortfolioMetadataResponse`

**Review by DavidLHW — COMMENTED** [review]
LGTM! aside from some minor comments that should be addressed

File: src/main/java/com/backend/controller/PortfolioController.java (hunk 1 above)
Comment (jeromeandrewong) [reply src/main/java/com/backend/controller/PortfolioController.java]: resolved 😸  [870608e](https://github.com/is442oop/portfolio-analyzer-backend/pull/35/commits/870608ee218040d9255553e09c3974c0e18ed40f)

File: src/main/java/com/backend/response/UpdatePortfolioMetadata.java (hunk 2 above)
Comment (jeromeandrewong) [reply src/main/java/com/backend/response/UpdatePortfolioMetadata.java]: resolved 😸  [870608e](https://github.com/is442oop/portfolio-analyzer-backend/pull/35/commits/870608ee218040d9255553e09c3974c0e18ed40f)

**Review by DavidLHW — APPROVED** [review]
LGTM!

**Pull request merged**